packages/components/forms/src/BaseInput/BaseInput.styles.ts

```diff
@@ -1,100 +1,138 @@
-    border: `1px solid ${isInvalid ? tokens.red600 : tokens.gray300}`,
-    borderRadius: tokens.borderRadiusMedium,
-    color: tokens.gray700,
-    fontFamily: tokens.fontStackPrimary,
-    fontSize: tokens.fontSizeM,
-    lineHeight: tokens.lineHeightM,
-    padding: `10px ${tokens.spacingS}`,
-    margin: 0,
-    cursor: isDisabled ? 'not-allowed' : 'auto',
-    width: '100%',
-    zIndex: getZIndex({ isDisabled, isInvalid }),
-
-    // if the input is a textarea, the resize prop is applied and size should be ignored
-    ...(as === 'textarea' ? { resize } : getSizeStyles({ size })),
+type getInputStylesProps = Pick<
+  BaseInputInternalProps,
+  'as' | 'isDisabled' | 'isInvalid' | 'size' | 'resize'
+> & {
+  density?: Density;
+};
 
-    '&::placeholder': {
-      color: tokens.gray500,
+const getStyles = ({
+  as,
+  isDisabled,
+  isInvalid,
+  size,
+  resize,
+  density = 'low',
+}: getInputStylesProps) => {
+  const densityStyles = {
+    low: {
+      borderRadius: tokens.borderRadiusMedium,
+      lineHeight: tokens.lineHeightM,
+      fontSize: tokens.fontSizeM,
     },
-
-    '&:active, &:active:hover': {
-      borderColor: isInvalid
-        ? tokens.red600
-        : isDisabled
-        ? tokens.gray300
-        : tokens.blue600,
-      boxShadow: isInvalid
-        ? tokens.glowNegative
-        : isDisabled
-        ? 'none'
-        : tokens.glowPrimary,
+    high: {
+      borderRadius: tokens.borderRadiusSmall,
+      lineHeight: tokens.lineHeightS,
+      fontSize: tokens.fontSizeS,
     },
+  };
 
-    '&:focus': {
-      borderColor: isInvalid
-        ? tokens.red600
-        : isDisabled
-        ? tokens.gray300
-        : tokens.blue600,
-      boxShadow: isInvalid
-        ? tokens.glowNegative
-        : isDisabled
-        ? 'none'
-        : tokens.glowPrimary,
-    },
-  }),
+  return {
+    rootComponentWithIcon: css({
+      position: 'relative',
+      display: 'flex',
+      width: '100%',
+      zIndex: getZIndex({ isDisabled, isInvalid }),
+    }),
+    input: css({
+      outline: 'none',
+      boxShadow: tokens.insetBoxShadowDefault,
+      boxSizing: 'border-box',
+      backgroundColor: isDisabled ? tokens.gray100 : tokens.colorWhite,
+      border: `1px solid ${isInvalid ? tokens.red600 : tokens.gray300}`,
+      color: tokens.gray700,
+      fontFamily: tokens.fontStackPrimary,
+      margin: 0,
+      cursor: isDisabled ? 'not-allowed' : 'auto',
+      width: '100%',
+      zIndex: getZIndex({ isDisabled, isInvalid }),
+      ...densityStyles[density],
+
+      // if the input is a textarea, the resize prop is applied and size should be ignored
+      ...(as === 'textarea' ? { resize } : getSizeStyles({ size, density })),
+
+      '&::placeholder': {
+        color: tokens.gray500,
+      },
 
-  inputWithIcon: css({
-    paddingLeft: size === 'small' ? tokens.spacingXl : '38px',
-  }),
+      '&:active, &:active:hover': {
+        borderColor: isInvalid
```

Comment:
nitpick: i think with this nested inline ifs, the styles here become harder to read and understand. Maybe it would be good to use a helper method for returning invalid and disabled styles?